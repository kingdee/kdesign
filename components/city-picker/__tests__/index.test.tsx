import React from 'react'
import { mount, render } from 'enzyme'
import CityPicker, { CityPickerProps } from '../index'
import { commons, groups, cityList } from './data'
import mountTest from '../../../tests/shared/mountTest'

const defaultProps: Partial<CityPickerProps> = {
  commons,
  groups,
  cityList,
  title: 'title',
  showTitle: true,
}

describe('CityPicker', () => {
  // 1. mount test
  mountTest(CityPicker)

  // 2. render test
  it('renders correctly', () => {
    const wrapper = render(<CityPicker title={'test'} />)
    expect(wrapper).toMatchSnapshot()
  })

  // 3. render no child without errors
  it('renders correctly with no child', () => {
    expect(mount(<CityPicker />)).toMatchSnapshot()
  })

  // 4. render null or undefined without errors
  it('renders correctly with null or undefined', () => {
    expect(
      mount(
        <CityPicker>
          {null}
          {undefined}
        </CityPicker>,
      ),
    ).toMatchSnapshot()
  })

  // 5. displayName
  it('should have displayName static property', () => {
    const wrapper = mount(<CityPicker></CityPicker>)
    expect((wrapper.type() as any).displayName).toBe('CityPicker')
  })

  // 6. className
  it('className', function () {
    const wrapper = mount(<CityPicker {...defaultProps} className={'my-class'} inputClassName={'my-input-class'} />)

    // className
    expect(wrapper.find('.kd-city-picker-container').at(0).hasClass('my-class')).toBeTruthy()

    // inputClassName
    expect(wrapper.find('input').hasClass('my-input-class')).toBeTruthy()
  })

  // 7. style
  it('style', function () {
    const wrapper = mount(<CityPicker {...defaultProps} />)

    // style
    expect(wrapper.find('.kd-city-picker-container').prop('style')).toBeFalsy()
    wrapper.setProps({ style: { height: '50px' } })
    wrapper.update()
    expect(wrapper.find('.kd-city-picker-container').prop('style')).toEqual({ height: '50px' })
    expect(wrapper.find('.kd-city-picker-container')).toHaveStyle('height', '50px')

    // input style
    expect(wrapper.find('input').prop('style')).toBeFalsy()
    wrapper.setProps({ inputStyle: { height: '20px' } })
    wrapper.update()
    expect(wrapper.find('input').prop('style')).toEqual({ height: '20px' })
    expect(wrapper.find('input')).toHaveStyle('height', '20px')
  })

  // 8. other api
  it('other api', () => {
    const wrapper = mount(<CityPicker {...defaultProps} />)

    // title
    expect(wrapper.find('.kd-city-picker-title').text()).toBe('title')

    // placeholder
    expect(wrapper.find('input').prop('placeholder')).toBeFalsy()
    wrapper.setProps({ placeholder: 'placeholder' })
    wrapper.update()
    expect(wrapper.find('input').props().placeholder).toBe('placeholder')

    // disabled
    expect(wrapper.find('input').prop('readOnly')).toBeFalsy()
    wrapper.setProps({ disabled: true })
    wrapper.update()
    expect(wrapper.find('input').prop('readOnly')).toBeTruthy()
  })
})
