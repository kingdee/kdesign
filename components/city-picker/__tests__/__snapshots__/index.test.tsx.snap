// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityPicker renders correctly 1`] = `
<div
  class="kd-city-picker-container"
>
  <div
    class="kd-city-picker-content"
  >
    <input
      class="kd-city-picker-input"
      title=""
      type="text"
      value=""
    />
    <span
      class="kd-city-picker-info"
    />
    <i
      class="kdicon kdicon-arrow-down kd-city-picker-arrow"
    />
  </div>
</div>
`;

exports[`CityPicker renders correctly with no child 1`] = `
<CityPicker>
  <div
    className="kd-city-picker-container"
    onClick={[Function]}
  >
    <div
      className="kd-city-picker-content"
    >
      <input
        className="kd-city-picker-input"
        onChange={[Function]}
        onFocus={[Function]}
        title=""
        type="text"
        value=""
      />
      <span
        className="kd-city-picker-info"
      />
      <i
        className="kdicon kdicon-arrow-down kd-city-picker-arrow"
      />
    </div>
  </div>
</CityPicker>
`;

exports[`CityPicker renders correctly with null or undefined 1`] = `
<CityPicker>
  <div
    className="kd-city-picker-container"
    onClick={[Function]}
  >
    <div
      className="kd-city-picker-content"
    >
      <input
        className="kd-city-picker-input"
        onChange={[Function]}
        onFocus={[Function]}
        title=""
        type="text"
        value=""
      />
      <span
        className="kd-city-picker-info"
      />
      <i
        className="kdicon kdicon-arrow-down kd-city-picker-arrow"
      />
    </div>
  </div>
</CityPicker>
`;
